website/docs/plugin/terraform-plugin-protocol.mdx

```diff
@@ -0,0 +1,227 @@
+
+| Protocol RPCs                                                                     | Framework Functionality                                                                                                                                            |
+|-----------------------------------------------------------------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `GetProviderSchema`                                                               | `provider.Provider`, `resource.Resource`, and `datasource.DataSource` interface `Schema` method                                                                    |
+| `ValidateProviderConfig` / `ValidateResourceConfig` /`ValidateDataResourceConfig` | `provider.Provider`, `resource.Resource`, and `datasource.DataSource` interface `Schema` method validators, `ConfigValidators` method, and `ValidateConfig` method |
+| `ConfigureProvider`                                                               | `provider.Provider` interface `Configure` method                                                                                                                   |
+| `ReadResource` / `ReadDataSource`                                                 | `resource.Resource` and `datasource.DataSource` interface `Read` method                                                                                            |
+| `PlanResourceChange`                                                              | `resource.Resource` interface `Schema` method plan modifiers and `ModifyPlan` method                                                                               |
+
+### _terraform apply_
+
+Executing _terraform apply_ (with default -refresh=true and -auto-approve) results in the following unique RPCs:
+
+![diagram: Terraform Plugin Protocol 6 Plan RPCs](/img/docs/terraform-plugin-protocol-6-rpcs-apply.png)
+
+The actual sequence of RPCs is as follows:
+
+- [GetProviderSchema](#getproviderschema-rpc)
+- [GetProviderSchema](#getproviderschema-rpc)
+- [ValidateProviderConfig](#validateconfig-rpcs)
+- [ValidateResourceConfig](#validateconfig-rpcs)
+- [ValidateDataResourceConfig](#validateconfig-rpcs)
+- [GetProviderSchema](#getproviderschema-rpc)
+- [ValidateProviderConfig](#validateconfig-rpcs)
+- [ConfigureProvider](#configureprovider-rpc)
+- UpgradeResourceState
+  - only called if resource already exists in state
+- [ValidateResourceConfig](#validateconfig-rpcs)
+- [ValidateDataResourceConfig](#validateconfig-rpcs)
+- [ReadResource](#read-rpcs)
+  - only called if resource already exists in state)
+- [ReadDataSource](#read-rpcs)
+- [PlanResourceChange](#planresourcechange-rpc)
+  - if _-auto-approve_ is not used this is the point at which Terraform prompts the practitioner to confirm the proposed changes
```

Comment:
We may want to confer with the core team to determine if they want to document this level of detail. For provider developers, I believe the important details are which RPCs are potentially called.

Reply:
I'll remove this for now.